go-apps/meep-app-enablement/server/app-support/app-support.go

```diff
@@ -0,0 +1,314 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package server
+
+import (
+	"time"
+
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	msmgmt "github.com/InterDigitalInc/AdvantEDGE/go-apps/meep-app-enablement/server/service-mgmt"
+	dkm "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-data-key-mgr"
+	log "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-logger"
+	redis "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-redis"
+
+	"github.com/gorilla/mux"
+)
+
+const mappsupportBasePath = "/mec_app_support/v1/"
+const mappsupportKey = "mec-app-support"
+const appEnablementKey = "app-enablement"
+const ACTIVE = "ACTIVE"
+const INACTIVE = "INACTIVE"
+
+//const logModuleMSMgmt = "meep-app-enablement"
+//const serviceName = "MEC Service Management"
+
+var redisAddr string = "meep-redis-master.default.svc.cluster.local:6379"
+
+//var influxAddr string = "http://meep-influxdb.default.svc.cluster.local:8086"
+
+var APP_ENABLEMENT_DB = 5
+
+var rc *redis.Connector
+var hostUrl *url.URL
+var sandboxName string
+var selfName string
+var basePath string
+var baseKey string
+var appEnablementBaseKey string
+
+var expiryTicker *time.Ticker
+
+var nextSubscriptionIdAvailable int
+
+/*func notImplemented(w http.ResponseWriter, r *http.Request) {
+        w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+        w.WriteHeader(http.StatusNotImplemented)
+}
+*/
+func Init() (err error) {
+	// Retrieve Sandbox name from environment variable
+	sandboxNameEnv := strings.TrimSpace(os.Getenv("MEEP_SANDBOX_NAME"))
+	if sandboxNameEnv != "" {
+		sandboxName = sandboxNameEnv
+	}
+	if sandboxName == "" {
+		err = errors.New("MEEP_SANDBOX_NAME env variable not set")
+		log.Error(err.Error())
+		return err
+	}
+	log.Info("MEEP_SANDBOX_NAME: ", sandboxName)
+
+	// hostUrl is the url of the node serving the resourceURL
+	// Retrieve public url address where service is reachable, if not present, use Host URL environment variable
+	hostUrl, err = url.Parse(strings.TrimSpace(os.Getenv("MEEP_PUBLIC_URL")))
+	if err != nil || hostUrl == nil || hostUrl.String() == "" {
+		hostUrl, err = url.Parse(strings.TrimSpace(os.Getenv("MEEP_HOST_URL")))
+		if err != nil {
+			hostUrl = new(url.URL)
+		}
+	}
+	log.Info("resource URL: ", hostUrl)
+
+	selfNameEnv := strings.TrimSpace(os.Getenv("MEEP_SELF_NAME"))
+	if selfNameEnv != "" {
+		selfName = selfNameEnv
+	}
+	//TODO
+	/*
+		if selfName == "" {
+			err = errors.New("MEEP_SELF_NAME env variable not set")
+			log.Error(err.Error())
+			return err
+		}
+	*/
+	//HARDCODE SOMETHING
+	selfName = "mep1"
+	log.Info("MEEP_SELF_NAME: ", selfName)
+
+	// Set base path
+	basePath = "/" + sandboxName + mappsupportBasePath
+	// Get base store key
+	baseKey = dkm.GetKeyRoot(sandboxName) + mappsupportKey
```

Comment:
not removed because it is actually used in my next task (subscriptions registrations...) so i left it there rather than having to put it back again...